x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/components/dashboard_enablement_panel.tsx

```diff
@@ -0,0 +1,199 @@
+        setModalState({ visible: false });
+      }
+    },
+    [storeEnablement, initRiskEngine]
+  );
+
+  if (storeEnablement.error) {
+    return (
+      <>
+        <EuiCallOut
+          title={
+            <FormattedMessage
+              id="xpack.securitySolution.entityAnalytics.entityStore.enablement.mutation.errorTitle"
+              defaultMessage={'There was a problem initializing the entity store'}
+            />
+          }
+          color="danger"
+          iconType="error"
+        >
+          <p>{(storeEnablement.error as { body: { message: string } }).body.message}</p>
+        </EuiCallOut>
+      </>
+    );
+  }
+
+  if (riskEngineInitializing) {
+    return (
+      <EuiPanel hasBorder data-test-subj="riskEngineInitializingPanel">
+        <EuiEmptyPrompt
+          icon={<EuiLoadingLogo logo="logoElastic" size="xl" />}
+          title={<h2>{ENABLEMENT_INITIALIZING_RISK_ENGINE}</h2>}
+        />
+      </EuiPanel>
+    );
+  }
+
+  if (entityStoreStatus === 'installing' || storeEnablement.isLoading) {
+    return (
+      <EuiPanel hasBorder data-test-subj="entityStoreInitializingPanel">
+        <EuiEmptyPrompt
+          icon={<EuiLoadingLogo logo="logoElastic" size="xl" />}
+          title={<h2>{ENABLEMENT_INITIALIZING_ENTITY_STORE}</h2>}
+          body={
+            <p>
+              <FormattedMessage
+                id="xpack.securitySolution.entityAnalytics.entityStore.enablement.initializing.description"
+                defaultMessage="This can take up to 5 minutes."
+              />
+            </p>
+          }
+        />
+      </EuiPanel>
+    );
+  }
+
+  if (
+    riskEngineStatus !== RiskEngineStatusEnum.NOT_INSTALLED &&
+    (entityStoreStatus === 'running' || entityStoreStatus === 'stopped')
+  ) {
+    return null;
+  }
+
+  const [title, body] = getEnablementTexts(entityStoreStatus, riskEngineStatus);
+  return (
+    <>
+      <EuiEmptyPrompt
+        css={{ minWidth: '100%' }}
+        hasBorder
+        layout="horizontal"
+        className="eui-fullWidth"
+        title={<h2>{title}</h2>}
+        body={<p>{body}</p>}
+        actions={
+          <EuiToolTip content={title}>
+            <EuiButton
+              color="primary"
+              fill
+              onClick={() => setModalState({ visible: true })}
+              data-test-subj={`entityStoreEnablementButton`}
+            >
+              <FormattedMessage
+                id="xpack.securitySolution.entityAnalytics.entityStore.enablement.enableButton"
+                defaultMessage="Enable"
+              />
+            </EuiButton>
+          </EuiToolTip>
+        }
+        icon={<EuiImage size="l" hasShadow src={dashboardEnableImg} alt={title} />}
+        data-test-subj="entityStoreEnablementPanel"
+      />
+
+      <EntityStoreEnablementModal
+        visible={modal.visible}
+        toggle={(visible) => setModalState({ visible })}
+        enableStore={enableEntityStore}
+        riskScore={{
```

Comment:
Not 100% but it might be neater to give the `EntityStoreEnablementModal` the responsibility of deciding what is checked and disbaled ,and just passing the raw statuses into that component?

Reply:
Idk... I think we're gonna change this anyways, we've talked about just removing the toggle if it's disabled